### PR TITLE
Revert "boards: 96b_carbon: Enable NET_L2_BT by default"

### PR DESCRIPTION
**Pull request by loicpoulain**
This reverts commit 6019bcaee58fd1362a18a32763d7ccadfa991a0d.

Bluetooth is no more the only network interface since Network
over usb is now operational (CDC-ECM). This fix network samples
with CDC-ECM where bluetooth is considered as the default iface.

### CODE REVIEW

**Review by pfalcon — CHANGES_REQUESTED** [review]
It's good that BT is no longer the only interface. Arguably, it still should be the default, because that's the functionality of the board. Otherwise any, absolutely any, board can use SLIP over serial, and by this logic, there can't be a default network interface set for any board. (Which is at least doesn't correspond to the real situation, as other boards keep having a default after this patch; actually, 96b_carbon was set a default on example of other boards).

Summing up, please find another way to fix CDC-ECM samples without breaking generic samples (what this patch almost certainly does).

P.S. It's also helpful to notify the original author of a patch if it's being reverted.

**Comment by Vudentz** [comment]
@jukkar Isn't there a way to configure the default interface? I guess that would be needed if there are multiple interfaces active on the system.

@pfalcon I agree that it is not a good idea to start disabling things just for the sake of making a sample behave properly, the sample should have means to detect what interfaces are available and switch to whatever interface it wants to use.

**Comment by loicpoulain** [comment]
@pfalcon, you're right, this patch is breaking the current default sample behavior for 96 carbon, because no more iface will then be available by default. What is disturbing for me is that with a same sample config, enabled features will be different depending the board (even the similar ones, BT boards...). I expected that board Kconfig only enables hw related capability symbols.

Today, enabling networking on 96b_carbon automatically selects the Bluetooth stack and make it the default iface even if BT is not voluntary selected.

After all, maybe this is more a network/sample issue and the way iface is selected.

P.S. yes, my bad, I thought you were automatically notified by github in case of revert.

**Comment by pfalcon** [comment]
@Vudentz , @loicpoulain : To remind, the motion in that direction wasn't started by me, but by @nashif instead. It was spread over several commits, but here's example: b92733d47a69cf6d0fac523ef48775ab23507cdb . The idea behind it is: for any network sample, there should be the "default" prj.conf, which should build and work for any board support networking. To make that work like that, there should be designated a default network interface for each board (in its config). Apparently, the criteria for the selection should be: a) it's a builtin hardware; b) among builtin features, the most prominent, most interesting to users feature should be selected.

I hope all so far makes sense - we can't make users wonder how to build a particular network sample for their board, not having a gazillion on board-specific prj*.conf's scale. It all should just work out of the box, to offer users a smooth starting curve with Zephyr.

But then how to support non-default network config? Well, one would need to *disable* the default interface (potentially, disable features it depends on, BT in this case), then enable something else. Sorry, but that's the price to pay for a smooth start described above. The only alternative is to make a user dig into board-specific options right from the start, then repeat for each new board. Instead, we offer an easy start to let a user rejoice that it works and motivate to study the configuration process next.

Finally, we can add non-default configs in-tree for easy start too, let it be e.g. prj_96b_carbon_usb_eth.conf. Per @nashif work, that shouldn't be spread around, just choose one sample (echo_server/echo_client are canonical ones for that).

Again, hope it all makes sense, if not, we need to open a separate ticket for discussion and extending/adjusting existing requirements.

**Comment by loicpoulain** [comment]
@pfalcon, thanks for explanation. It makes the sample configs generic/scalable by absorbing 'complexity' in each board config. It indeed makes sense from user perspective, so I assume I can live with that.
